tests/glm/test_utils.py

```diff
@@ -96,3 +96,52 @@ def test_align_df_categories_missing_columns(df):
 def test_align_df_categories_not_df():
     with pytest.raises(TypeError):
         _align_df_categories(np.array([[0], [1]]), {"x0": np.float64})
+
+
+@pytest.fixture()
+def df_na():
```

Comment:
Would be good to have tests for the different cases mentioned above: a data frame with a categorical column that already contains `"(M)"` and a data frame that does not contain all dtypes.